### PR TITLE
Use `MatmulType` to check for right element type configuration of matmul.

### PR DESCRIPTION
**Pull request by MaheshRavishankar**
This is followup for comments in #12085.

### CODE REVIEW

**Review by bjacob — APPROVED** [review]
I will take care of adapting the existing SetEncoding.cpp code to make use of this new getMatmulType function.

File: compiler/src/iree/compiler/Codegen/Common/EncodingInfo.cpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+                                        Type rhsElementType,
+                                        Type resultElementType) {
+  if (lhsElementType.isSignlessInteger(8) &&
+      rhsElementType.isSignlessInteger(8) &&
+      resultElementType.isSignlessInteger(32)) {
+    return MatmulType::I8I8I32;
+  } else if (lhsElementType.isF32() && rhsElementType.isF32() &&
+             resultElementType.isF32()) {
+    return MatmulType::F32F32F32;
+  }
```
Comment (hanhanW) [inline compiler/src/iree/compiler/Codegen/Common/EncodingInfo.cpp:24]: IIRC, there is a bot that suggested me now using `else if` when it's returning a result. (Sorry that I don't have a reference now.)

```suggestion
  }
  if (lhsElementType.isF32() && rhsElementType.isF32() &&
             resultElementType.isF32()) {
    return MatmulType::F32F32F32;
  }
```

**Review by hanhanW — APPROVED** [review]

**Comment by MaheshRavishankar** [comment]
Actually this is causing a circular dependency. I am punting on this for now.

**Comment by bjacob** [comment]
No problem, I will take this over.

**Comment by hanhanW** [comment]
> No problem, I will take this over.

It's fixed in https://github.com/iree-org/iree/pull/12182

**Comment by bjacob** [comment]
So fast! Thanks!

**Pull request closed without merging**